packages/langium/src/validation/document-validator.ts

```diff
@@ -236,13 +236,13 @@ export function getDiagnosticRange<N extends AstNode>(info: DiagnosticInfo<N, st
 export function toDiagnosticSeverity(severity: LexingDiagnosticSeverity): DiagnosticSeverity {
     switch (severity) {
         case 'error':
-            return 1; // according to vscode-languageserver-types/lib/esm/main.js#DiagnosticSeverity.Error
+            return 1 satisfies typeof DiagnosticSeverity.Error;
         case 'warning':
-            return 2; // according to vscode-languageserver-types/lib/esm/main.js#DiagnosticSeverity.Warning
+            return 2 satisfies typeof DiagnosticSeverity.Warning;
         case 'info':
-            return 3; // according to vscode-languageserver-types/lib/esm/main.js#DiagnosticSeverity.Information
+            return 3 satisfies typeof DiagnosticSeverity.Information;
         case 'hint':
-            return 4; // according to vscode-languageserver-types/lib/esm/main.js#DiagnosticSeverity.Hint
+            return 4 satisfies typeof DiagnosticSeverity.Hint;
```

Comment:
Nice, at first I read this and wasn't sure about being so specific, but honestly it looks pretty good. My only thought would be to retain at least one comment (somewhere) denoting `according to vscode-languageserver-types/lib/esm/main.js#DiagnosticSeverity`. It may be obvious via the type, but retaining the information might help for others are looking at it later.

Reply:
Specific is good :D
But it's mainly about type safety and readability, to guarantee that copy-pasting (and forgetting to change the number) errors won't happen.

> retain at least one comment

I don't understand why such a comment would be needed. `DiagnosticSeverity` comes directly from that package, which can be jumped to using a simple "go to definition" (<kbd>CTRL/⌘</kbd> + click</kbd>)

Reply:
Good comments can be surprisingly helpful, even when they appear redundant. The types are also excellent, but I would always error on the side of being verbose here. To be more specific, the comment _reinforces_ the developer's intent, and the type _agrees_ with that intent. This can make it easier to check later on if something changes (or needs to change).

Reply:
Sounds good. The function was even missing a JSDoc comment, so it seems like a good place to add it

Reply:
Perfect, thank you 😄